com.woltlab.wcf/templates/membersList.tpl

```diff
@@ -1,172 +1,25 @@
-{capture assign='pageTitle'}{if $searchID}{lang}wcf.user.search.results{/lang}{else}{$__wcf->getActivePage()->getTitle()}{/if}{if $pageNo > 1} - {lang}wcf.page.pageNo{/lang}{/if}{/capture}
-
-{capture assign='contentTitle'}{if $searchID}{lang}wcf.user.search.results{/lang}{else}{$__wcf->getActivePage()->getTitle()}{/if} <span class="badge">{#$items}</span>{/capture}
-
-{capture assign='canonicalURLParameters'}sortField={$sortField}&sortOrder={$sortOrder}{if $letter}&letter={unsafe:$letter|rawurlencode}{/if}{/capture}
+{if $searchID}
+	{capture assign='pageTitle'}{lang}wcf.user.search.results{/lang}{/capture}
+	{capture assign='contentTitle'}{lang}wcf.user.search.results{/lang}{/capture}
+{/if}
 
 {capture assign='headContent'}
-	{if $pageNo < $pages}
-		<link rel="next" href="{link controller='MembersList' pageNo=$pageNo+1}{unsafe:$canonicalURLParameters}{/link}">
+	{if $listView->getPageNo() < $listView->countPages()}
+		<link rel="next" href="{link controller='MembersList'}pageNo={$listView->getPageNo() + 1}{/link}">
 	{/if}
-	{if $pageNo > 1}
-		<link rel="prev" href="{link controller='MembersList'}{if $pageNo > 2}pageNo={$pageNo-1}&{/if}{unsafe:$canonicalURLParameters}{/link}">
+	{if $listView->getPageNo() > 1}
+		<link rel="prev" href="{link controller='MembersList'}{if $listView->getPageNo() > 2}pageNo={$listView->getPageNo() - 1}{/if}{/link}">
 	{/if}
```

Comment:
This is a repeated pattern and it is an annoying piece of copy & paste. In addition, the `pageNo` parameter is hidden from the request route.

We could solve this by introducing two new helper methods to `AbstractListView()`:

- `public function getNextPageNo(): ?int`
- `public function getPreviousPageNo(): ?int`

They essentially do the same but can now be added as an actual parameter. Also the `{if}` checks can be turned into a simple equality check again `null`.